### PR TITLE
add basic support for OS X

### PR DESCRIPTION
**Pull request by digitalmoksha**

### CODE REVIEW

File: lib/motion-fabric.rb [hunk 1]
```diff
@@ -90,16 +93,26 @@ def fabric(&block)
...
-  # "use_frameworks!" option from Cocoapods.
-  app.framework_search_paths << './vendor/Pods/Fabric/iOS'
+  if osx?
+    # staticlly link the frameworks
+    app.vendor_project('./vendor/Pods/Fabric/OSX/Fabric.framework', :static, products: ['Fabric'])
+    app.vendor_project('./vendor/Pods/Crashlytics/OSX/Crashlytics.framework', :static, products: ['Crashlytics'])
+
+    # make sure they are not copied into the Frameworks folder
+    app.embedded_frameworks.delete_if {|item| item.to_s.include?('Fabric.framework')}
+    app.embedded_frameworks.delete_if {|item| item.to_s.include?('Crashlytics.framework')}
```
Comment (MarkVillacampa) [inline lib/motion-fabric.rb:103]: Why is this needed? We're not manipulating the `app.embedded_frameworks` array.

File: lib/motion-fabric.rb [hunk 2]
```diff
@@ -26,6 +26,9 @@
 unless defined?(Motion::Project::Config)
   raise "This file must be required within a RubyMotion project Rakefile."
 end
+def osx?
+  App.config_without_setup.deploy_platform == 'MacOSX'
```
Comment (MarkVillacampa) [inline lib/motion-fabric.rb]: I'd recommend you use `App.template == :osx` instead of this

**Comment by MarkVillacampa** [comment]
This looks great!

I've left a couple comments. If you can answer them and fix the conflicts I'll merge :)

File: lib/motion-fabric.rb (hunk 2 above)
Comment (digitalmoksha) [reply lib/motion-fabric.rb]: yep you're right, fixed

File: lib/motion-fabric.rb (hunk 1 above)
Comment (digitalmoksha) [reply lib/motion-fabric.rb:103]: If I don't do that, I get these warnings:

```
  WARNING! Cannot locate and fix install name path of embedded framework `/Users/walkerbl/Sites/In_Development/Versatil/versatil/macos/vendor/Pods/Crashlytics/OSX/Crashlytics.framework' in executable `./build/MacOSX-10.10-Release/Versatil Markdown.app/Contents/MacOS/Versatil Markdown', application might not start
  WARNING! Cannot locate and fix install name path of embedded framework `/Users/walkerbl/Sites/In_Development/Versatil/versatil/macos/vendor/Pods/Fabric/OSX/Fabric.framework' in executable `./build/MacOSX-10.10-Release/Versatil Markdown.app/Contents/MacOS/Versatil Markdown', application might not start
```

This seemed the best way to solve it without some major surgery.  But I'm happy if there is a cleaner way.

Comment (MarkVillacampa) [reply lib/motion-fabric.rb:103]: Ahh I see what's going on. For OSX, motion-cocoapods assumes frameworks are real frameworks, not fake frameworks (meaning real shared libraries, instead of static libraries) so it adds them to `embedded_frameworks`:

https://github.com/HipByte/motion-cocoapods/blob/master/lib/motion/project/cocoapods.rb#L196

Your fix looks good 👍

**Comment by MarkVillacampa** [comment]
Can you rebase your changes on top of master?

**Comment by digitalmoksha** [comment]
I've done the rebase

**Comment by MarkVillacampa** [comment]
Thanks!
